src/main/java/com/denizenscript/depenizen/bukkit/bridges/LuckPermsBridge.java

```diff
@@ -1,66 +1,75 @@
-        TagManager.registerTagHandler(new TagRunnable.RootForm() {
-            @Override
-            public void run(ReplaceableTagEvent event) {
-                tagEvent(event);
-            }
-        }, "luckperms");
-        ObjectFetcher.registerWithObjectFetcher(LuckPermsTrackTag.class);
-        PropertyParser.registerProperty(LuckPermsPlayerProperties.class, PlayerTag.class);
-    }
+        public static LuckPermsBridgeTags instance;
 
-    public void tagEvent(ReplaceableTagEvent event) {
-        Attribute attribute = event.getAttributes().fulfill(1);
-
-        // <--[tag]
-        // @attribute <luckperms.list_tracks>
-        // @returns ListTag(LuckPermsTrackTag)
-        // @plugin Depenizen, LuckPerms
-        // @description
-        // Returns a list of all tracks.
-        // -->
-        if (attribute.startsWith("list_tracks")) {
-            ListTag tracks = new ListTag();
-            for (Track track : luckPermsInstance.getTrackManager().getLoadedTracks()) {
-                tracks.addObject(new LuckPermsTrackTag(track));
-            }
-            event.setReplacedObject(tracks.getObjectAttribute(attribute.fulfill(1)));
+        public LuckPermsBridgeTags() {
+            instance = this;
+            TagManager.registerStaticTagBaseHandler(LuckPermsBridgeTags.class, "luckperms", (t) -> instance);
         }
 
-        // <--[tag]
-        // @attribute <luckperms.track[<track_name>]>
-        // @returns LuckPermsTrackTag
-        // @plugin Depenizen, LuckPerms
-        // @description
-        // Returns the track from the name given.
-        // -->
-        if (attribute.startsWith("track")) {
-            if (attribute.hasParam()) {
-                event.setReplacedObject(attribute.paramAsType(LuckPermsTrackTag.class).getObjectAttribute(attribute.fulfill(1)));
-            }
+        public void register() {
+
+            // <--[tag]
+            // @attribute <luckperms.list_tracks>
+            // @returns ListTag(LuckPermsTrackTag)
+            // @plugin Depenizen, LuckPerms
+            // @description
+            // Returns a list of all tracks.
+            // -->
+            tagProcessor.registerTag(ListTag.class, "list_tracks", (attribute, object) -> {
+                return new ListTag(luckPermsInstance.getTrackManager().getLoadedTracks(), LuckPermsTrackTag::new);
+            });
+
+            // <--[tag]
+            // @attribute <luckperms.track[<track_name>]>
+            // @returns LuckPermsTrackTag
+            // @plugin Depenizen, LuckPerms
+            // @description
+            // Returns the track from the name given.
+            // -->
+            tagProcessor.registerTag(LuckPermsTrackTag.class, ElementTag.class, "track", (attribute, object, name) -> {
+                Track track = luckPermsInstance.getTrackManager().getTrack(name.toString());
+                return track != null ? new LuckPermsTrackTag(track) : null;
+            });
```

Comment:
Look at the previous code handling this - you don't need to do the conversion manually as `LuckPermsTrackTag` is already constructed from the track name, just make the param type `LuckPermsTrackTag` and return that